### ISSUE TITLE
connections/connection.js test errors

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#2649]
When the `connections/connection.js` test is run on MacOS the "Javascript connection API With a server" test is failing to run because the method it is using to detect the startup of the rehtinkdb server is failing, and even seems tone causing the server to never complete setup.

These problems do not seem to happen on newton, and it is unknown what causes the difference at this point.

The problems:

1.  In `test/rql_test/connections/connection.js` the lines with `fs.write(server_err_log, data)` are apparently causing a silent error, and need to be changed to `fs.write(server_err_log, data.toString())`. A similar change also needs to be made in the `cpp_server.stdout.on` handler.

2. If you fix issue 1, and then disable the `spawn('rm', ['-rf', 'rethinkdb_data']);` line in `afterEach`, and then look at the output in `run/server-error-log.txt` you will see a lot of lines like this:
```Previous message may not have been written to the log file (cannot flush stderr: Operation not supported).```
Obviously somewhere in the stack something is not working as expected. But it seems to me that outputting a message complaining the STDERR is not working as you expect on STDERR is particularly unhelpful[1]. Better would be to output this sort of message to syslog. Or at worst this message should only be printed once, and not at 'info'. If you bypass the PIPEs in the test, and redirect straight to files, then these lines go away, but that prevents the startup detection from working.

3. As long as the PIPEs are in place, nothing is ever written to `rethinkdb_data/log_file`. Removing the PIPEs fixes this problem, implying that when there is a problem flushing data to STDERR that writing data to the file is skipped.

To reproduce (on MacOS, such as Gantz):
```
git checkout next
./configure --allow-fetch
make DEBUG=1
cd test/rql_test
./test-runner connections/connection -i js
```

[1] Very much in the tradition of "No keyboard detected, press any key to continue".

**Comment by larkost** [comment]
After looking at this again I think that the simpler thing to do to fix this test is to not have it spin up its own server but rather use the server that `test-runner` is already offering. For all but one subtest this should work great. The two downsides:

1. If a subtest were to mess up the server it will not re-launch for other subtests. Since we don't report on sub-tests this is probably ok (and highlights the overloaded nature of the connection tests).
2. There is one subtest that needs to kill the server to check that we subsequently get the proper "connection closed" error. This subtest will not survive this change.

I talked with @danielmewes, and we are both ok with these downsides, so I am going to make this change. I will keep the more expansive changes above in mind for later if we need them.

**Comment by larkost** [comment]
This change is up for review in CR 2773.

**Comment by larkost** [comment]
This has been merged into `next` (68a85d5) and `v2.0.x` (f652ad1).
